### PR TITLE
feat: distinct VPC host prep client-landing-zone

### PR DESCRIPTION
**Pull request by alaincormier-ssc**
in preparation for feature #883

**NOTICE**: setters have been modified

**`client-landing-zone`** package:
  - DNS
    - add new DNS project.
    - copy/update the host-project `public-dns.yaml` resources into it, however they will be placed in comment for easier transition.  DNS resources for the client will be deployed in this project instead of the host project. The same zone can't exist twice, and a zone can't be deleted if it contains recordset(s).
    - add a notice in the host-project public-dns.yaml that it will be transitioned.
  - VPC host project
    - add new `nonp` and `pbmm` host projects, keeping only appropriate subnets.
    - label resources under `client-folder/standard/applications-infrastructure/host-project` as "legacy" for post-transition cleanup.
  - Firewall policies rules:
    - label "standard" and "applications-infrastructure" folder firewall-policy as "legacy" for post-transition cleanup.
    - replicate the `client-folder/standard/applications-infrastructure/firewall-policy` resources under the pbmm AND nonp folders (adjusting resource names and references accordingly).

### CODE REVIEW

File: solutions/client-landing-zone/client-folder/standard/applications-infrastructure/dns-project/project.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#########
+# Project
+apiVersion: resourcemanager.cnrm.cloud.google.com/v1beta1
+kind: Project
+metadata:
+  name: dns-project-id # kpt-set: ${dns-project-id}
+  namespace: client1erhm-projects # kpt-set: ${client-name}-projects
```
[the remark this answers is not in the document]
Comment (alaincormier-ssc) [reply solutions/client-landing-zone/client-folder/standard/applications-infrastructure/dns-project/project.yaml]: good catch, cleaned up in this file and the README

File: solutions/client-landing-zone/setters.yaml [hunk 2]
```diff
@@ -170,6 +176,10 @@ data:
   #
   # dns project id created during core-landing-zone package deployment
   # customization: required, obtain value from the core-landing-zone setters.yaml
+  core-dns-project-id: dns-project-12345
```
Comment (lucstjean-ssc) [inline solutions/client-landing-zone/setters.yaml:179]: should we add core in front of dns-project-12345 -> core-dns-project-12345?

Comment (alaincormier-ssc) [reply solutions/client-landing-zone/setters.yaml]: good idea
update pushed

File: solutions/client-landing-zone/client-folder/standard/applications-infrastructure/nonp/firewall-policy/rules/defaults.yaml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+# Rules 2147483541 to 2147483546 are the suggested defaults by Google
+#########
+# Delegate to host project, egress traffic(firewall) from VPC resources to private IP ranges in shared VPC network
+# AC-3(9), AC-4, AC-4(21), SC-7(5), SC-7(8), SC-7(9), SC-7(11) - All connections to or from virtual machine instances are allowed/denied via firewall rules configured in shared VPC network within host project or firewall policies in parent folders based on least-privilege principle. Each firewall rule applies to incoming(ingress) or outgoing(egress) connections, not both.
+# AU-12 - Enable Logging for firewall policies
+# SI-4 - Logging denied traffic
+apiVersion: compute.cnrm.cloud.google.com/v1beta1
+kind: ComputeFirewallPolicyRule
+metadata:
+  name: client-name-standard-app-infra-nonp-fwpol-exclude-private-ip-ranges-egress-fwr # kpt-set: ${client-name}-standard-app-infra-nonp-fwpol-exclude-private-ip-ranges-egress-fwr
```
Comment (lucstjean-ssc) [inline solutions/client-landing-zone/client-folder/standard/applications-infrastructure/nonp/firewall-policy/rules/defaults.yaml:24]: Have you tested these? Without the rendered client-name, client-name-standard-app-infra-nonp-fwpol-exclude-private-ip-ranges-egress-fwr is 78 characters long.

Comment (alaincormier-ssc) [reply solutions/client-landing-zone/client-folder/standard/applications-infrastructure/nonp/firewall-policy/rules/defaults.yaml:24]: yes I tested it and it deployed ok (I purposely shortened "applications-infrastructure" to "app-infra" in the name to keep it as short as possible)
for example, that same rule already exists on the "applications-infrastructure" folder and is 91 characters before rendering

**Review by davelanglois-ssc — APPROVED** [review]

**Review by fmichaelobrien — APPROVED** [review]
/lgtm

**Pull request merged**